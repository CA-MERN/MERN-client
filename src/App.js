import { connect } from "react-redux";
import React, { Fragment } from "react";
import Login from "./components/login";
import Register from "./components/register";
import UserSettings from "./components/userSettings";
import Preferences from "./components/preferences";
import SearchRecipeButton from "./components/resuables/searchButton";
import Nav from "./components/nav";
import "./App.css";
import Modal from "react-modal";
import styles from "./components/styles/app.module.css";


const customStyles = {
  content: {
    top: "50%",
    left: "50%",
    right: "auto",
    bottom: "auto",
    marginRight: "-50%",
    transform: "translate(-50%, -50%)",
  },
};
// const user = false;

const App = ({ userLoggedIn }) => {
  const modals = {
    register: <Register />,
    login: <Login />,
  };
  const [modalOpen, setModalOpen] = React.useState(null);
  const [modalIsOpen, setIsOpen] = React.useState(false);
  const openModal = () => {
    setIsOpen(true);
  };
    
  function closeModal() {
    setIsOpen(false);
  }

  
  return (
    <div className={styles.homeLayoutOnly}>
           <Nav />
          <main className={` ${styles.homeContent}`}>
              <div className={styles.homeBox}>
                  <p>Random Food Jokes API</p>
                  <h1>What is in your fridge?</h1>
                  <h3>Just add your ingredients and FridgeMate will help find recipes personlised to you!</h3>
                  <SearchRecipeButton/>
                  <h4>
                    {userLoggedIn ? "you are logged in" : "please log in you loser"}
                  </h4>
                  <div>
                      {userLoggedIn ? (
                  <Fragment>
                    <UserSettings /> <Preferences />

                  </Fragment>
                ) : (
                  <Fragment>
                    <button
                      onClick={() => {
                        setModalOpen("login");
                        openModal();
                      }}
                    >
                      Login
                    </button>
                    <button
                      onClick={() => {
                        setModalOpen("register");
                        openModal();
                      }}
                    >
                      Register
                    </button>
              
                  </Fragment>
                )}
                <Modal
                  isOpen={modalIsOpen}
                  onRequestClose={closeModal}
                  style={customStyles}
                  ariaHideApp={false}
                  contentLabel="Example Modal"
                >
                  {modalOpen && modals[modalOpen]}
                  <button
                    class="modalLoginButton"
                    onClick={() => {
                      setModalOpen("login");
                      openModal();
                    }}
                  >
                    Login
                  </button>
                  <button
                    class="modalRegisterButton"
                    onClick={() => {
                      setModalOpen("register");
                      openModal();
                    }}
                  >
                    Register
                  </button>
                  <button
                    class="modalCancelButton"
                    onClick={() => {
                      closeModal();
                    }}
                  >
                    Cancel
                  </button>
                </Modal>
              </div>
          
              </div>
     
         </main>



      
    </div>
  );
};

const mapStateToProps = (state) => ({
  userLoggedIn: state.userLoggedIn.username,
});

export default connect(mapStateToProps)(App);

// routes for later

// HOME
// /

// SIGN IN PAGE
// user/login

// REGISTER PAGE
// user/register

// ACCOUNT SETTINGS PAGE
// user/:username/account-settings

// PREFRENCES PAGE
// preferences/:username

// FRIDGE PAGE
// ingredients/:username/fridge

// PANTRY PAGE
// ingredients/:username/pantry

// BROWSE RECIPE PAGE
// /recipes/browse

// SINGLE RECIPE
// /recipes/:recipe-name

// USER SAVED RECIPES
// /saved-recipes/:username

// also You need to add the massive ingredient JSON to your client repo when you do the autocmplete haha. If you pull the latest from server then you can move the file to a data file in the client :)
// https://material-ui.com/components/autocomplete/

// user: {
//   fridgeIngredients: [Array],
//   pantryIngredients: [Array],
//   _id: '5fd0136d3d8569a20f8504b0',
//   name: 'Test User 1',
//   email: 'tester@test.com',
//   username: 'testusername',
//   password: '$2a$10$n7gAww9x3nTlZFYQQ25zHuu4S8ix6U44akrTo7pP5GlqOba6bOWtK',
//   createdDate: '2020-12-08T23:59:41.384Z',
//   lastLogin: '2020-12-08T23:59:41.384Z',
//   __v: 0,
//   profile: 'https://fridgemate.s3.ap-southeast-2.amazonaws.com/1607471981565'
//   }
