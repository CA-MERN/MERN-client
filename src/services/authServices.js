import api from '../config/api'



export async function registerUser(userInfo) {
    // call to server to register user
    const response = await api.post("/user/register", userInfo)
    console.log("got user back from server", response)
    return response.data
}

export async function loginUser(userInfo) {
    // call to server to login user
    // return user info if successful and error if not
    const response = await api.post("/user/login", userInfo)
    console.log("got user back from server", response) 
   // console.log(response) 
    return response.data
}

export async function logoutUser() {
    // call to server to logout user
    return api.get("/user/logout")
}

export async function getUserSettings(username) {
    // call to server to register user
    const response = await api.get("/user/"+username+"/account-settings")
    console.log("got user back from server", response)
    return response.data
}  


export async function updateUserSettings(settingInfo,username) {
    // call to server to register user
    const response = await api.patch("/user/"+username+"/account-settings", settingInfo)
    console.log("got user back from server", response)
    return response.data
}  


export async function uploadProfileImage(image,username) {
    // call to server to register user
    const response = await api.post("/user/"+username+"/add-profile-picture", image)
    console.log("got user back from server", response)
    return response.data
}  


export async function updatePreference(userInfo,username) {
    // call to server to register user
    const response = await api.post("/preferences/"+username+"/edit", userInfo)
    console.log("got user back from server", response)
    return response.data
}  

//Right now we are saving the token - but as token is in a cookie this does not need to be in local storage- remove later
// Get loggedInUser from localStorage
export function getLoggedInUser() {
   return localStorage.getItem("token")
}

// Store loggedInUser token in local storage
export function setLoggedInUser(token) {
    token ? localStorage.setItem("token", token) : localStorage.removeItem("token")
    console.log("local token Updated")
    console.log(token)
}



// Get loggedInUser from localStorage
export function getUsername() {
    return localStorage.getItem("username")
}

// Store loggedInUser username in local storage
export function setUsername(user) {
    user ? localStorage.setItem("username", user) : localStorage.removeItem("username")
    console.log("local username Updated")
    console.log(user)
}