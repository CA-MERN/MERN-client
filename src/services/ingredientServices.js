import api from '../config/api'

//FRIDGE

// Returns all Fridge ingredients
export async function getAllFridgeIngredients(username) {
    const response = await api.get("/ingredients/"+username+"/fridge")
    return response.data
}

// Adds a fridge item on the server
export async function addFridgeItem(username,item) {
    const response = await api.post("/ingredients/"+username+"/fridge/new", item)
    return response.data
}

// Deletes a single fridge item on the server
export async function deleteFridgeItem(username,item) {
    const response = await api.delete("/ingredients/"+username+"/fridge/delete", item)
    return response.data
}

// Clears all fridge items from server
export async function deleteAllFridge(username) {
    const response = await api.delete("/ingredients/"+username+"/fridge/clear-all")
    return response.data
}

//PANTRY

// Returns all Pantry ingredients
export async function getAllPantryIngredients(username) {
    const response = await api.get("/ingredients/"+username+"/pantry")
    return response.data
}

// Adds a pantry item on the server
export async function addPantryItem(username,item) {
    const response = await api.post("/ingredients/"+username+"/pantry/new", item)
    return response.data
}

// Deletes a single pantry item on the server
export async function deletePantryItem(username,item) {
    const response = await api.delete("/ingredients/"+username+"/pantry/delete", item)
    return response.data
}

// Clears all pantry items from server
export async function deleteAllPantry(username,item) {
    const response = await api.delete("/ingredients/"+username+"/pantry/clear-all", item)
    return response.data
}




export function getFridge() {
    return localStorage.getItem("fridge")
}


export function setFridge(items) {
    items ? localStorage.setItem("fridge", items) : localStorage.removeItem("fridge")
    console.log("local fridge Updated")
    console.log(items)
}

export function getPantry() {
    return localStorage.getItem("pantry")
}

export function setPantry(user) {
    user ? localStorage.setItem("pantry", user) : localStorage.removeItem("pantry")
    console.log("local pantry Updated")
    console.log(user)
}