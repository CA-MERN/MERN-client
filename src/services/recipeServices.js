import api from '../config/api'


// Returns Recipes to browse
export async function browseSearchRecipes() {
    const response = await api.get("/recipes/browse")
     await setBrowsedRecipes(response.data)
     console.log("Server", response);
    return response.data
}

// Returns all user saved recipes
export async function getAllUserSavedRecipes() {
    const response = await api.get("/recipes/saved-recipes")
    return response.data
}

// Get Single Recipe Page
export async function getSingleRecipePage(recipeId) {
    const response = await api.get("/recipes/"+recipeId)
    //console.log("Server", response);
    return response.data
}

// Add to User Saved Recipes
export async function addNewSavedRecipe(recipe) {
    const response = await api.post("/recipes/add", recipe)
    return response.data
}

// Delete to User Saved Recipes
export async function removedSavedRecipe(recipeId) {
    const response = await api.delete(`/recipes/${recipeId}`);
    return response.data
}


export function getBrowsedRecipes() {
    return localStorage.getItem("browsedRecipes")
}

export function setBrowsedRecipes(items) {
    items ? localStorage.setItem("browsedRecipes", JSON.stringify(items)) : localStorage.removeItem("browsedRecipes")
    console.log("local browsedRecipes Updated")
    //console.log(items)
}

export function getSavedRecipes() {
    return localStorage.getItem("savedRecipes")
}

export function setSavedRecipes(recipe) {
    recipe ? localStorage.setItem("savedRecipes", JSON.stringify(recipe)) : localStorage.removeItem("savedRecipes")
    console.log("local savedRecipes Updated")
    //console.log(recipe)
}


export function getSingleRecipe() {
    return localStorage.getItem("singleRecipe")
}

export function setSingleRecipe(recipe) {
    recipe ? localStorage.setItem("singleRecipe", JSON.stringify(recipe)) : localStorage.removeItem("singleRecipe")
    console.log("local singleRecipe Updated")
    //console.log(recipe)
}
