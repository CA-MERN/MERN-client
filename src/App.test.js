import { shallow } from "enzyme";
import React from "react";

import App from "./App";
import { getLoggedInUser } from "./services/authServices";

jest.mock("./services/authServices");

test("renders learn react link", () => {
  expect(shallow(<App />)).toMatchSnapshot();
});

it("Throws error when attempting to log in", () => {
  getLoggedInUser.mockImplementation(() => {
    throw new Error();
  });
  expect(shallow(<App />)).toMatchSnapshot();
});
