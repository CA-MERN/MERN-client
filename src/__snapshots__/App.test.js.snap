// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      FridgeMate
    </title>
    <meta
      content="Helmet application"
      name="description"
    />
  </HelmetWrapper>
  <WithStyles(CssBaseline) />
  <BrowserRouter>
    <Connect(NavBar) />
    <Switch>
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
          }
        }
        exact={true}
        path="/"
      />
      <PrivateRoute
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
          }
        }
        exact={true}
        path="/preferences/:username"
      />
      <PrivateRoute
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
          }
        }
        exact={true}
        path="/user/:username/account-settings"
      />
      <PrivateRoute
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
          }
        }
        exact={true}
        path="/ingredients/:username/fridge"
      />
      <PrivateRoute
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
          }
        }
        exact={true}
        path="/recipes/browse"
      />
      <PrivateRoute
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
          }
        }
        exact={true}
        path="/recipes/:id/recipe"
      />
      <PrivateRoute
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
          }
        }
        exact={true}
        path="/recipes/saved-recipes"
      />
      <PrivateRoute
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
          }
        }
        exact={true}
        path="/ingredients/:username/pantry"
      />
      <PrivateRoute
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
          }
        }
        exact={true}
        path="/user/reset-password/:token"
      />
      <Route
        component={[Function]}
      />
    </Switch>
    <ToastContainer
      autoClose={5000}
      closeButton={[Function]}
      closeOnClick={true}
      draggable={true}
      draggablePercent={80}
      hideProgressBar={false}
      newestOnTop={false}
      pauseOnFocusLoss={true}
      pauseOnHover={true}
      position="top-right"
      role="alert"
      rtl={false}
      transition={[Function]}
    />
  </BrowserRouter>
  <Footer />
</Fragment>
`;
